### ISSUE TITLE
message notifications appear not until I open TS

### ISSUE THREAD

**Issue by tomorilla** [WhisperSystems/TextSecure#3111]
I don't get notifications for received messages until i open TS. (no sound, no vibrating, no LED, ...) 
No matter if:
- TS is locked/unlocked
- display is on/off
- message received in a group or not

I only use push messages.

I have a Sony Xperia Z1C with Android 5.0.2. 
I think this issue appeared for the first time after i upgraded from KitKat to Lollipop. (but I'm not absolutly sure)

I have two friends with the same problem. Both have Android 5. (one of them has a Sony Xperia Z2 and the other has a Nexus)

**Comment by moxie0** [comment]
This means you either:

1) Have background data disabled for google services.

2) Have background data disabled for TS.

3) Have a bad gapps install.

4) Have used something like Titanium Backup.

5) Have a network that is blocking GCM messages.

**Comment by ahamrani** [comment]
Hello,

I have the same issue with my Honor 3C european version, with Jelly bean 4.2.2. I think the Emotion Ui 2.0 from Huawei has something to do with it because it's too resctrictive. I gave all authorizations to textSecure, but the problem still. Also, in the main screen, where all contacts are listed, the names does not appears properly, it appears every second with the phone number. Maybe textSecure tries to access to contact list saved in the phone but it the emotion Ui doesn't give access ?

Here a debug log : https://gist.github.com/anonymous/12c1994c1a9fb6cf7b9a

Thanks for your help.

**Comment by rhodey** [comment]
@ahamrani unfortunately this log did not capture any interesting behavior, next time this delayed notification situation happens could you try and immediately capture and submit a debug log?

**Comment by rehans** [comment]
I had the same problem. GCM and GSF are a real PITA. In order to make TextSecure run correctly with GCM I did as follows:

1. Factory reset my phone
2. Install GApps of my phone manufacturer (an "older" version for 4.2.2)
3. Install TextSecure 
4. Start TS and activate Push. Activation fails cause of outdated GSF.
5. Goto Google Play and update GSF
6. Activate Push, this time it does not complain about outdated GSF. Registration succeeds.
7. Look at the Wifi Icon in Android status bar. It stays grey (no connection to GCM), not matter what I do.
8. Goto App Manager and deinstall GSF update.
9. Restart phone and look at the Wifi icon again. It becomes blue (connected to GCM) and TS works as expected.

In case you unregister from Push I need to follow all these steps again (except step 1.). 

Probably this "workaround" is helpfull.

**Comment by rhodey** [comment]
@rehans thanks for the detailed write up :D
 
I'm going to close this as resolved and also cc: @jlund in here because I expect your write up might be useful for our [support center](support.whispersystems.org)

**Comment by ahamrani** [comment]
Hello,

I'm really sorry for the answer delay, I had some issues with my computer, I didn't try rehans solution, I will this evening.

I have created a new issue topic for the other issue that I've mentioned with a new debug Log [here](https://github.com/WhisperSystems/TextSecure/issues/3376).

Many thanks.

**Comment by behrmann** [comment]
I seem to have this problem intermittently on a CyanogenMod 12.1 nightly (from last Wednesday) on my Bacon. I had deactivated some Google Apps (among them the Google Search App) and did not receive push notifications anymore, after activating the Google Search App again, push notifications resumed, only to to stop working again on Friday. Just before writing this the update to TS 2.24.1 hit me and push notifications seem to work again. I'll wait whether the problem returns an will provide a debug log in that case.

**Comment by JavaJens** [comment]
I also have this issue since a couple of days.
The debug log doesn't seem suspicious, but what is interesting that GCM DID receive the message.

I checked the GCM Event Log by dialing ```*#*#426#*#*``` and saw that GCM reported a message for TextSecure, however the message only arrived in TextSecure once I opened it.

**Comment by m-ad** [comment]
> 4) Have used something like Titanium Backup.

Can someone elaborate on why Titanium Backup might cause this behaviour?

I experience this bug on a rooted stock ROM MotoG (1st Gen), Android 4.4.4, TS 2.25.3. Background data is enabled for GoogleServices and TS. It worked before, so I guess gapps and my network are not to blame.

**Comment by JavaJens** [comment]
@m-ad The idea is that you also backup your GCM registration number which is linked to your device. If you now do a fresh install of your phone you also get a new registration number. When you now use a backup with the old registration, your messages are sent to the nirvana as your phone doesn't check for these messages.
It is only when you open Textsecure that a direct connection to the servers, regardless of GCM, is made and you thus receive your messages.

**Comment by quietsche** [comment]
@JavaJens So how to fix this? Having also this problem with TS and getting the messages when opening the app. Threema/WA/etc working fine. Had to do a full restore with TWRP due to Google Play Services problem, experiencing problems since then. Already deleted and reinstalled TS, additional to the message de-/re-registering.

**Comment by JavaJens** [comment]
@quietsche Re-registering should solve that problem. I guess you could unregister, wipe all the data of TS and then install+register again. But not 100% this works, as the problem I described above should already be solved by re-registering.

**Comment by quietsche** [comment]
Re-Registering did not solve the problem. Having some strange logs in google services framework log. Any idea what this CANCELED is?
![google play services log](http://picpaste.de/pics/IYCzTdXw.1444379979.jpg)

**Comment by jedie** [comment]
Same as: #4266

**Comment by encrus** [comment]
- I have background data enabled (anyway I am on wifi)
- I have checked and GCM is working (used GCM test app)
- I have no firewall restrictions to google app, google play, google services
- Signal notifications ARE not working.
    MY friend has the same problem. Unless the app is running, notifications are not working

Moxie: thanks so much for giving this app. But its use is being hugely limited due to lack of notifications working properly!
Kindly fix this or help us fix it. 

If this can't be fixed. OR please provide a setting to start the app upon reboot and keep this ON by default (preferably) but at least option should be there.

**Comment by quietsche** [comment]
For me it was fixed as I enabled the Signal app to be active in standby in Xperia Stamina mode.

**Comment by dkerlee** [comment]
Enabled Settings - Apps - Signal - Data Usage - "unrestricted data usage" 

I had Data Saver turned on. 

Android 7.1.1
